docs/source/plug-in.md

```diff
@@ -31,7 +31,7 @@ class TemplatePersonality(LoggingConfigurable):
         been specified."""
         pass
 
-def create_personality(self, parent):
+def create_personality(parent, log):
```

Comment:
Good catch!  

Should the signature be: `def create_personality(*args, **kwargs):` and we document two required kwargs are: `parent` and `log`?  This will then fit the existing declarations that reside for the [websocket](https://github.com/jupyter-server/kernel_gateway/blob/c72803ab1ea4bb75847917e0a7c682acbe700fbe/kernel_gateway/jupyter_websocket/__init__.py#L81) and [http](https://github.com/jupyter-server/kernel_gateway/blob/c72803ab1ea4bb75847917e0a7c682acbe700fbe/kernel_gateway/notebook_http/__init__.py#L156) personalities while allowing for other parameters to be added in the future.

Reply:
I'm unclear about the usage of `*args, **kwargs` here: Based on [`init_configurables`](https://github.com/jupyter-server/kernel_gateway/blob/c72803ab1ea4bb75847917e0a7c682acbe700fbe/kernel_gateway/gatewayapp.py#L447), I'm not I'm not seeing how any additional args could be passed to `create_personality` via configuration.

Reply:
In the future, we may find the need to convey other information to a "personality", at which time we'd add a parameter (e.g., `foo=bar`) in `init_configurables`.  Personalities that don't know about the parameter could still be called (and ignore it), while those that know what `foo` is for would use it via `kwargs["foo"]`, for example.

This, coupled with the idea that signatures for "custom personalities" should match that of the two existing personalities, leads me to argue for using `*args, **kwargs` due to precedent.

One of the things that makes this confusing is the lack of initializers with "Configurable" classes - on which most of Jupyter is based.  Both `parent` and `log` are "well-known" parameters within the `LoggingConfigurable` class hierarchy.  (Not sure if this helps or not.)

Reply:
Makes sense to use that for convention and extensibility.